test/common/heap.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable node-core/required-modules */
+'use strict';
+const assert = require('assert');
+const util = require('util');
+const { createJSHeapDump, buildEmbedderGraph } = require('internal/test/heap');
+
+class State {
+  constructor() {
+    this.snapshot = createJSHeapDump();
+    this.embedderGraph = buildEmbedderGraph();
+  }
+
+  validateSnapshotNodes(name, expected, { loose = false } = {}) {
+    const snapshot = this.snapshot.filter(
+      (node) => node.name === name && node.type !== 'string');
+    if (loose)
+      assert(snapshot.length >= expected.length);
+    else
+      assert.strictEqual(snapshot.length, expected.length);
+    for (const expectedNode of expected) {
+      if (expectedNode.edges) {
+        for (const expectedEdge of expectedNode.edges) {
+          const check = typeof expectedEdge === 'function' ?
+            expectedEdge : (node) => node.name === expectedEdge.name;
```

Comment:
Yup, sounds good. Done!

While addressing this, it occurred to me that `expectedChild` may or may not be a sensible variable name to use (because of the real-world usage of that term), but I’ll leave it that way until somebody has something better. :)